extensions/amp-video/1.0/video-wrapper.js

```diff
@@ -92,17 +101,35 @@ export function VideoWrapper({
   // <source>s change.
   const readyDeferred = useMemo(() => new Deferred(), []);
 
-  const play = useCallback(() => {
-    readyDeferred.promise.then(() => {
-      playerRef.current.play();
-    });
-  }, [readyDeferred]);
+  const play = useCallback(
+    () =>
+      readyDeferred.promise.then(() => {
+        const couldBePromise = playerRef.current.play();
+        if (couldBePromise && couldBePromise.catch) {
+          couldBePromise.catch(() => {
+            // Empty catch to prevent useless unhandled rejection logging.
```

Comment:
I see the reason, but do you think this is a right layer though? Maybe AMP Element should ignore this error in `registerAction`. But as an imperative API it might be better to still propagate it to the caller.

Similarly, I think a disconnected promise has the same question. I'd prefer if we returned it here and let the caller deal with it.

Reply:
Putting it here does feel weird. I'm not sure `registerAction` would be right, since `play()` can be called internally via autoplay, etc.

Alternatives:

- `onPlaybackError` prop.
- Catching only `MEDIA_ERR_*`.

Reply:
Autoplay is totally ok to suppress that error, since it's a special callsite for the `play` method. Thus the idea is to have `play():Promise` to be as clean as possible. And each caller decides what to do with failures.

Reply:
Works for me.